src/btree/bt_cursor.c

```diff
@@ -307,13 +304,20 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint64_t recno, bool *vali
      */
     if (cbt->ins != NULL) {
         if (WT_CURSOR_BOUNDS_SET(&cbt->iface)) {
-            /* Get the insert list key. */
-            if (key == NULL && btree->type == BTREE_ROW) {
-                tmp_key.data = WT_INSERT_KEY(cbt->ins);
-                tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
-            }
-            WT_RET(__btcur_bounds_contains_key(
-              session, &cbt->iface, key != NULL ? key : &tmp_key, &key_out_of_bounds, NULL));
+            if (btree->type == BTREE_ROW) {
+                /* Get the insert list key. */
+                if (key == NULL) {
+                    tmp_key.data = WT_INSERT_KEY(cbt->ins);
+                    tmp_key.size = WT_INSERT_KEY_SIZE(cbt->ins);
+                    WT_RET(__btcur_bounds_contains_key(
+                      session, &cbt->iface, &tmp_key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
+                } else
+                    WT_RET(__btcur_bounds_contains_key(
+                      session, &cbt->iface, key, WT_RECNO_OOB, &key_out_of_bounds, NULL));
+            } else
+                WT_RET(__btcur_bounds_contains_key(
+                  session, &cbt->iface, NULL, cbt->recno, &key_out_of_bounds, NULL));
```

Comment:
When are we using `cbt->recno ` as vs `cursor->recno ` ? Seems a little confusing to me that we're getting the recno from different places and it doesn't seem to be documented anywhere how we choose it.

Reply:
Found out there's a ticket made for this already: https://jira.mongodb.org/browse/WT-9699

Reply:
That ticket is a little different from the comment. The ticket describes a new way of being able to retrieve the key for those functions. We should be more clear with what cbt->recno vs cursor->recno, I'll do some investigation to make sure this is right then I'll merge once we are more clear.

Reply:
Thanks!